qgis_deployment_toolbelt/jobs/job_qgis_installation_finder.py

```diff
@@ -0,0 +1,306 @@
+            for filename in os.listdir(bin_dir):
+                if binary_pattern.match(filename):
+                    qgis_exe = os.path.join(bin_dir, filename)
+                    return qgis_exe
+        return None
+
+    @staticmethod
+    def _get_windows_installed_qgis_path() -> dict[str, str]:
+        """Get dict of installed QGIS version in common install directory
+
+        Returns:
+            dict[str, str]: dict of QGIS version and QGIS bin path
+        """
+        # Check for installed version in the default install directory
+        found_version = {}
+
+        # Program files
+        prog_file_dir = expandvars("%PROGRAMFILES%")
+        directory_pattern = re.compile(r"QGIS (\d+)\.(\d+)\.(\d+)", re.IGNORECASE)
+        for dir_name in os.listdir(prog_file_dir):
+            # Check if the directory name matches the pattern
+            match = directory_pattern.match(dir_name)
+            if match:
+                install_dir = os.path.join(prog_file_dir, dir_name)
+                if qgis_bin := JobQgisInstallationFinder._get_qgis_bin_in_install_dir(
+                    install_dir
+                ):
+                    version_str = JobQgisInstallationFinder._get_qgis_bin_version(
+                        qgis_bin=qgis_bin
+                    )
+                    if version_str:
+                        found_version[version_str] = qgis_bin
+                    else:
+                        logger.warning(
+                            f"Can't define QGIS version for '{qgis_bin}' binary."
+                        )
+
+        # OSGEO4W
+        install_dir = os.environ.get("QDT_OSGEO4W_INSTALL_DIR", "C:\\OSGeo4W")
+        if qgis_bin := JobQgisInstallationFinder._get_qgis_bin_in_install_dir(
+            install_dir
+        ):
+            version_str = JobQgisInstallationFinder._get_qgis_bin_version(
+                qgis_bin=qgis_bin
+            )
+            if version_str:
+                found_version[version_str] = qgis_bin
+            else:
+                logger.warning(f"Can't define QGIS version for '{qgis_bin}' binary.")
+
+        return found_version
+
+    @staticmethod
+    def _get_linux_installed_qgis_path() -> dict[str, str]:
+        """Get install qgis path for linux operating system with which
+
+        Returns:
+            dict[str, str]: dict of QGIS version and QGIS bin path
+        """
+        # use which to find installed qgis
+        found_version = {}
+        if qgis_bin := which("qgis"):
+            logger.debug(f"QGIS path found using which: {qgis_bin}")
+            version_str = JobQgisInstallationFinder._get_qgis_bin_version(
+                qgis_bin=qgis_bin
+            )
+            if version_str:
+                found_version[version_str] = qgis_bin
+            else:
+                logger.warning(f"Can't define QGIS version for '{qgis_bin}' binary.")
+
+        return found_version
```

Comment:
It looks like a code duplication from [constant](https://github.com/Guts/qgis-deployment-cli/blob/main/qgis_deployment_toolbelt/constants.py#L199)